### PR TITLE
fix: fix for oui-datagrid in metrics

### PR DESCRIPTION
**Pull request by varun257**
### Description of the Change

replace table with datagrid
replace search with datagrid search

### Benefits

standard oui-kit datagrid

### Possible Drawbacks

None

### Applicable Issues

SCDC-754

### CODE REVIEW

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 1]
```diff
@@ -4,46 +4,44 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
+            <oui-column title="'metrics_token_labels' | translate" type="string" property="labels">
+                <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key'" data-ng-if="$index < 2">
+                    <span data-ng-bind="label.key"></span>:<span data-ng-bind="label.value"></span>
+                </span>
+                <oui-dropdown data-ng-if="$row.labels.length > 2">
+                    <div class="token-labels__button-container">
+                        <button type="button" class="token-labels token-labels__button" data-ng-bind="MetricsTokenCtrl.displayRemainingLabels($row.labels.length - 2)" oui-dropdown-trigger></button>
+                    </div>
+                    <oui-dropdown-content class="token-labels-container">
+                        <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key'" data-ng-if="$index >= 2">
```
Comment (FredericEspiau) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `track by`

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 2]
```diff
@@ -4,46 +4,44 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
+            <oui-column title="'metrics_table_name' | translate" sortable="asc" searchable property="description" type="string">
+                    <span data-ng-bind="$row.description || '-'"></span>
+            </oui-column>
+            <oui-column title="'metrics_token_labels' | translate" type="string" property="labels">
+                <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key'" data-ng-if="$index < 2">
+                    <span data-ng-bind="label.key"></span>:<span data-ng-bind="label.value"></span>
+                </span>
+                <oui-dropdown data-ng-if="$row.labels.length > 2">
+                    <div class="token-labels__button-container">
+                        <button type="button" class="token-labels token-labels__button" data-ng-bind="MetricsTokenCtrl.displayRemainingLabels($row.labels.length - 2)" oui-dropdown-trigger></button>
```
Comment (FredericEspiau) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `oui-dropdown-trigger` ?

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 3]
```diff
@@ -4,46 +4,44 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
-            <column title="'metrics_token_permission' | translate"
-                    property="permission">
+        <oui-datagrid rows="MetricsTokenCtrl.tokens">
+            <extra-top>
+                <button class="oui-button oui-button_secondary" type="button"
+                    data-ng-disabled="MetricsTokenCtrl.actions.loading"
+                    data-translate="metrics_token_add_token"
+                    ui-sref="dbaas.metrics.detail.token.add({ serviceName: MetricsTokenCtrl.serviceName })"></button>
+            </extra-top>
+            <oui-column title="'metrics_table_name' | translate" sortable="asc" searchable property="description" type="string">
```
Comment (FredericEspiau) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `data-`

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html (hunk 2 above)
Comment (varun257) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.html]: Yup, there is a dropdown if thecount of labels is more than 2

Comment (FredericEspiau) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.html]: Alright, I didn't know about `oui-dropdown-trigger` 😅

**Comment by varun257** [comment]
@FredericEspiau have made the changes. Please have a look

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 4]
```diff
@@ -4,46 +4,45 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
-                <cui-clipboard data-text="$row.access"></cui-clipboard>
-            </column>
-            <column title="'metrics_token_permission' | translate"
-                    property="permission">
+        <oui-datagrid rows="MetricsTokenCtrl.tokens">
+            <extra-top>
+                <button class="oui-button oui-button_secondary" type="button"
+                    data-ng-disabled="MetricsTokenCtrl.actions.loading"
+                    data-translate="metrics_token_add_token"
+                    ui-sref="dbaas.metrics.detail.token.add({ serviceName: MetricsTokenCtrl.serviceName })"></button>
```
Comment (JeremyDec) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `data-`

Comment (FredericEspiau) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `data-`

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 5]
```diff
@@ -4,46 +4,45 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
+            </extra-top>
+            <oui-column data-title="'metrics_table_name' | translate" data-sortable="asc" data-searchable data-property="description" data-type="string">
```
Comment (marie-j) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `::`on `data-title`

**Comment by antleblanc** [comment]
@varun257 Could you please update/set labels?

File: package.json [hunk 6]
```diff
@@ -90,12 +90,13 @@
     "@bower_components/ovh-manager-webfont": "ovh-ux/ovh-manager-webfont#^1.0.1",
     "@bower_components/uri.js": "medialize/URI.js#~1.15.0",
     "@bower_components/validator-js": "chriso/validator.js#~3.37.0",
-    "ovh-ui-angular": "~2.15.0",
-    "ovh-ui-kit": "~2.15.0",
-    "popper.js": "^1.14.1"
+    "ovh-ui-angular": "^2.15.0",
+    "ovh-ui-kit": "^2.15.0",
+    "popper.js": "^1.14.1",
+    "serve": "^9.1.0"
```
Comment (Netnix) [inline package.json]: I think the changes in this file should be moved to another PR that explains what it does and how it improves the build.

Comment (antleblanc) [reply package.json]: Furthermore I have no idea why `"serve"` is installed. Does this dependency is required/used?

**Comment by antleblanc** [comment]
@varun257 there are conflicts :)

**Comment by jleveugle** [comment]
Hello @varun257,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

File: client/app/dbaas/dbaas-metrics/token/metrics-token.less [hunk 7]
```diff
@@ -47,6 +48,14 @@
     }
 
   }
+  &__row {
+    .oui-datagrid__row {
+      .oui-input-group_clipboard {
```
Comment (antleblanc) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.less]: Must be done in our OVH UI Kit (https://github.com/ovh-ux/ovh-ui-kit/tree/master/packages/oui-datagrid)

File: package.json [hunk 8]
```diff
@@ -96,7 +96,7 @@
     "popper.js": "^1.14.1"
   },
   "devDependencies": {
-    "autoprefixer": "^6.0.0",
+    "autoprefixer": "^6.7.7",
```
Comment (antleblanc) [inline package.json]: Is there a specific reason for updating those dependencies (according to this pull request)?

File: client/app/dbaas/dbaas-metrics/token/metrics-token.less (hunk 7 above)
Comment (varun257) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.less]: This style is required only for this table as there are some customized columns in this which breaks the clipboard

File: package.json (hunk 8 above)
Comment (varun257) [reply package.json]: Have reverted these changes

File: client/app/dbaas/dbaas-metrics/token/metrics-token.less (hunk 7 above)
Comment (FredericEspiau) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.less]: Could be a case to talk about with @AxelPeter then 😃

Comment (AxelPeter) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.less]: Margins on this kind of control will be removed in the next releases. 
But why the flex ?

Comment (varun257) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.less]: The clipboard component has display block and this is actually breaking the table since the other column has variable height and produces an unnecessary scroll bar next to the clipboard column. Hence used display flex on the clipboards so that it fits.

**Comment by antleblanc** [comment]
@varun257 Is there any ticket related to this pull request? Also could you please resolve the conflict? Thanks:)

**Comment by varun257** [comment]
@antleblanc Have updated issue id and fixed merge conflicts. The css changes can be considered as temp fix for now :)

File: client/app/dbaas/dbaas-metrics/token/metrics-token.less (hunk 7 above)
Comment (antleblanc) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.less]: I personally prefer adding a small comment `TODO:…` and explain why we have this code and where we have to fix the root cause.

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 9]
```diff
@@ -5,45 +5,48 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
-                    property="labels"
-                    template="MetricsTokenCtrl.labelsTemplate()">
-            </column>
-            <column title="'metrics_table_token' | translate"
-                    property="access">
-                <oui-clipboard data-model="$row.access"></oui-clipboard>
-            </column>
-            <column title="'metrics_token_permission' | translate"
-                    property="permission">
+        <oui-datagrid rows="MetricsTokenCtrl.tokens">
```
Comment (antleblanc) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: +`data-`

File: client/app/dbaas/dbaas-metrics/token/metrics-token.html [hunk 10]
```diff
@@ -5,45 +5,48 @@ <h3 class="oui-header_3" data-translate="metrics_token_title"></h3>
...
+                </span>
+                <oui-dropdown data-ng-if="$row.labels.length > 2">
+                    <button type="button"
+                        class="token-labels token-labels__button"
+                        data-ng-bind="MetricsTokenCtrl.displayRemainingLabels($row.labels.length - 2)"
+                        oui-dropdown-trigger>
+                    </button>
+                    <oui-dropdown-content>
+                        <div class="token-labels-container">
+                            <span class="token-labels" data-ng-repeat="label in $row.labels | orderBy : 'key' track by $index" data-ng-if="$index >= 2">
```
Comment (antleblanc) [inline client/app/dbaas/dbaas-metrics/token/metrics-token.html]: I'm not sure about what a `token-labels` looks like but why you didn't use `chips` components? (ref: http://master.ui-kit.ovh/#!/oui-angular/chips)

Comment (cbourgois) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.html]: 👍

Comment (varun257) [reply client/app/dbaas/dbaas-metrics/token/metrics-token.html]: `token-labels` was a local class. Have removed it and added `chips` component with some customisation. Please have a look
